plugins/auto-sizes/optimization-detective.php

```diff
@@ -16,7 +16,7 @@
  * @since 1.1.0
  *
  * @param OD_Tag_Visitor_Context $context Tag visitor context.
- * @return false Whether the tag should be recorded in URL metrics.
+ * @return false Whether the tag should be tracked in URL metrics.
  */
 function auto_sizes_visit_tag( OD_Tag_Visitor_Context $context ): bool {
```

Comment:
@westonruter 
> which code bits are you referring to?

Haven't looked at the codebase outside this PR, but this function for example would better have a different name now. Of course this is just a plugin _consuming_ Optimization Detective, but it led me to bring this up.

But there's definitely also relevant code in Optimization Detective itself, e.g. the `od_register_tag_visitors` action could probably use a more accurate name too.

Reply:
These are still tag visitors though. That doesn't change here. What is changed is that tag visitors always visit tags (and they always have) but what is made more clear is what the return value means from a tag visitor. A true return value means that it is indicating that the tag should be tracked/registered/monitored in URL metrics. In the case of the Enhanced Responsive Images plugin, the tag visitor never needs to read the tag out of URL metrics so it always returns false.

Reply:
But why are they called "tag visitors"? I agree with the rationale for the changes in this PR, but in the same sense I find it equally unclear what "visiting" is about when looking at the API as a whole. "Tracking" a tag makes a lot more sense for me in describing the purpose of why I should be using this API. The purpose, as far as I understand, is to track certain tags - which implies "visiting", but that's not really the purpose and less clear to me.

Reply:
Tag visitors visit every tag in the document. 

Only a subset of tag visitors need to track the tag in URL metrics. If a tag visitor doesn't need URL metrics to perform it's optimizations (as is the case for auto-sizes) then no tracking is happening.

Reply:
Ah, thanks for clarifying!